guides/source/active_record_validations.md

```diff
@@ -1017,8 +1017,10 @@ instance.
 
 ```ruby
 class EmailValidator < ActiveModel::EachValidator
+  REGEX = /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i
+
   def validate_each(record, attribute, value)
-    unless value =~ /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i
+    unless value =~ REGEX
```

Comment:
This makes the example both longer and harder to read, in my opinion. What's the benefit of this change?

Reply:
I think we should encourage people to put a regex like this in a constant.
Isn't that the recommended approach (unless the regex contains interpolations)?
Sure, it makes the example longer, but in my opinion easier to read and understand.

Reply:
I also like constants in some situations, but my vote would be to remove the constant in this case. Perhaps you could use another topic to explain validators? Email validation is whole a topic in unto itself, and has many nuances. If that seems like too much of a hassle then the email could be changed to something simpler like [`.+\@.+\..+`](https://www.wired.com/2008/08/four-regular-expressions-to-check-email-addresses/), which would also fit your criteria of being  easier to read.

Reply:
Alright, @eugeneius would you accept to remove the constant and change the regex to something simple like `/.+@.+\..+/`?

Reply:
Ping @eugeneius! 😸 Would love to resolve this to get this PR merged!

Reply:
The goal of this section is to showcase the `EachValidator` API; the parts of the code that implement the specific validation should be as easy to skim over as possible. Extracting a constant may be a good idea in a real validator, but unnecessarily steals focus here. Switching to `.+@.+\..+` would reduce the amount of example-specific code... but it might be _too_ simple, and open a [can of worms](https://discuss.rubyonrails.org/t/should-rails-be-able-to-validate-email-out-of-the-box/74314) about how to write a "correct" email validator. I think the fact that the current regex is fairly impenetrable is a feature, to serve as a reminder that understanding it isn't the point.

Maybe switching to a different example entirely could be an improvement, but I still think this example is fine as-is.

Reply:
I agree, let's keep it as-is. 👍

Reply:
Updated!